drupal/web/sites/all/modules/costum/imdb movie info crawler/imdb_movie_info_crawler.module

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * @file
+ * This module crawls Imdb movie pages and displays the movie title, picture and movie plot. 
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function imdb_movie_info_crawler_menu() {
+    $items['Imdb_Crawler'] = array(
+      'title' => 'Imdb crawler',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('imdb_movie_info_crawler_form'),
+      'access callback' => TRUE,
+      'type' => MENU_NORMAL_ITEM,
+    );
+    $items['Movie_Info'] = array(
+      'title' => 'Movie Info',
+      'page callback' => 'movie_info_page_callback',
+      'access callback' => TRUE,
+      'type' => MENU_LOCAL_TASK,
+    );
+    return $items;
+}
+
+/**
+ * Page callback for Imdb_Crawler path.
+ */
+function imdb_movie_info_crawler_form($form, &$form_state) {
+    $form['movie_url'] = array(
+      '#title' => t('Enter Imdb movie link'),
+      '#type' => 'textfield',
+      '#description' => 'Enter the imdb movie link here',
+      '#size' => 32,
+      '#maxlength' => 128,
+      '#required' => TRUE,
+    );
+    $form['submit_button'] = array(
+      '#type' => 'submit',
+      '#value' => t('Submit!'),
+    );
+    return $form;
+}
+
+/**
+ * Implements hook_form_validate().
+ */
+function imdb_movie_info_crawler_form_validate($form, &$form_state) {
+    $submitted_url = $form_state['values']['movie_url'];
+    $string_pattern_url = "'http://www.imdb.com/title/tt'si";
+    $valid_url = preg_match($string_pattern_url, $submitted_url);
+    if (!$valid_url) {
+        drupal_set_message("Please enter a valid url!");
+        $form_state['rebuild'] = TRUE;
+    }
+}
+
+/**
+ * Implements hook_form_submit().
+ */
+function imdb_movie_info_crawler_form_submit($form, &$form_state) {
+    $movie_infos = get_imdb_movie_infos($form_state['values']['movie_url']);
+    $_SESSION['movie_infos'] = $movie_infos;
+    drupal_goto("Movie_Info");
```

Comment:
Hinweise:

1) drupal_goto('url', array('query' => array('title' => $movie_infos['title'], ))
2) SeSSION
3) write result to db -> redirect with ID -> on new page load from db

4) Goto() -> $form_state['redirect'] ersetzten.